src/libwalletqt/Subaddress.cpp

```diff
@@ -0,0 +1,56 @@
+#include "Subaddress.h"
+#include <QDebug>
+
+Subaddress::Subaddress(Monero::Subaddress *subaddressImpl, QObject *parent)
+  : QObject(parent), m_subaddressImpl(subaddressImpl)
+{
+    qDebug(__FUNCTION__);
+    getAll();
+}
+
+QList<Monero::SubaddressRow*> Subaddress::getAll(bool update) const
+{
+    qDebug(__FUNCTION__);
+
+    emit refreshStarted();
+
+    if(update)
+        m_rows.clear();
+
+    if (m_rows.empty()){
+        for (auto &row: m_subaddressImpl->getAll()) {
+            m_rows.append(row);
+        }
+    }
+
+    emit refreshFinished();
+    return m_rows;
+}
+
+Monero::SubaddressRow * Subaddress::getRow(int index) const
+{
+    return m_rows.at(index);
```

Comment:
Does this need a range test ?

Reply:
Range check was already done in the caller's side, but added a range test anyway.